src/Scripting/Core/ScriptCompiler.cs

```diff
@@ -13,7 +13,7 @@ internal abstract class ScriptCompiler
         public abstract DiagnosticFormatter DiagnosticFormatter { get; }
         public abstract StringComparer IdentifierComparer { get; }
 
-        public abstract SyntaxTree ParseSubmission(SourceText text, CancellationToken cancellationToken);
+        public abstract SyntaxTree ParseSubmission(SourceText text, ScriptOptions scriptOptions, CancellationToken cancellationToken);
```

Comment:
>ScriptOptions [](start = 68, length = 13)

I'd type this to `ParseOptions` since the parsing should not depend on other options.